frontend/src/stimulus/controllers/dynamic/openid-connect/match-preview-dialog.controller.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+import { TurboRequestsService } from 'core-app/core/turbo/turbo-requests.service';
+
+export interface MatchPreviewDialogSubmittedEvent {
+  detail:{
+    regularExpressions:string;
+    previewGroups:string;
```

Comment:
Do we use `previewGroups` data somewhere? If not then why do we have it as a part of event data?

Reply:
Indeed we don't use it. I was very much experimenting with Stimulus and thinking about what best practices might be worth establishing.

My thinking was that this dialog doesn't care what you do with its data, but it collects two inputs from the user, so it makes sense to also emit those collected inputs in the submitted event. I.e. the submitted event is generic and doesn't care about the receiver.

I guess the practical impact is low:

* You correctly point out, that it's not used; I also don't believe that we will start using it any time soon
* on the other hand, it also doesn't really hurt to have it in the event data (the event contains a lot of information nobody cares about too)